test_scripts/Defects/8_2/2468_1_AddCommand_Both_UI_Timeout.lua

```diff
@@ -0,0 +1,141 @@
+---------------------------------------------------------------------------------------------------
+
+--[[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('test_scripts/Smoke/commonSmoke')
+
+--[[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+--[[ Local Variables ]]
+local putFileParams = {
+  requestParams = {
+    syncFileName = 'icon.png',
+    fileType = "GRAPHIC_PNG",
+    persistentFile = false,
+    systemFile = false
+  },
+  filePath = "files/icon.png"
+}
+
+local requestParams = {
+  cmdID = 11,
+  menuParams = {
+    position = 0,
+    menuName = "Commandpositive",
+    secondaryText = "Secondary",
+    tertiaryText = "Tertiary"
+  },
+  vrCommands = {
+    "VRCommandonepositive",
+    "VRCommandonepositivedouble"
+  },
+  cmdIcon = {
+    value = "icon.png",
+    imageType = "DYNAMIC"
+  },
+  secondaryImage = {
+    value = "icon.png",
+    imageType = "DYNAMIC"
+  }
+}
+
+local responseUiParams = {
+  cmdID = requestParams.cmdID,
+  cmdIcon = requestParams.cmdIcon,
+  menuParams = requestParams.menuParams,
+  secondaryImage = requestParams.secondaryImage
+}
+
+local responseVrParams = {
+  cmdID = requestParams.cmdID,
+  type = "Command",
+  vrCommands = requestParams.vrCommands
+}
+
+local allParams = {
+  requestParams = requestParams,
+  responseUiParams = responseUiParams,
+  responseVrParams = responseVrParams
+}
+
+--[[ Local Functions ]]
+local function addCommand(pParams)
+  local cid = common.getMobileSession():SendRPC("AddCommand", pParams.requestParams)
+
+  pParams.responseUiParams.appID = common.getHMIAppId()
+  pParams.responseUiParams.cmdIcon.value = common.getPathToFileInAppStorage("icon.png")
+  pParams.responseUiParams.secondaryImage.value = common.getPathToFileInAppStorage("icon.png")
+  common.getHMIConnection():ExpectRequest("UI.AddCommand", pParams.responseUiParams)
+  :Do(function(_, data)
+      -- No UI response
+      -- common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+      common.getHMIConnection():ExpectRequest("UI.DeleteCommand", {
+        cmdID = pParams.requestParams.cmdID
+      }):Timeout(20000)
+    end)
+
+  pParams.responseVrParams.appID = common.getHMIAppId()
+  common.getHMIConnection():ExpectRequest("VR.AddCommand", pParams.responseVrParams)
+  :Do(function(_, data)
+      common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+      common.getHMIConnection():ExpectRequest("VR.DeleteCommand", { 
+        cmdID = pParams.requestParams.cmdID,
+        type = "Command",
+        grammarID = data.params.grammarID
+      }):Timeout(20000)
+    end)
+  :ValidIf(function(_, data)
+    if data.params.grammarID ~= nil then
+      return true
+    else
+      return false, "grammarID should not be empty"
+    end
+  end)
+
+  common.getMobileSession():ExpectResponse(cid, {
+    success = false,
+    resultCode = "GENERIC_ERROR"
+  }):Timeout(20000)
```

Comment:
A slight nitpick but currently the default timeout is default period + DefaultTimeoutCompensation (11s) so maybe 
```suggestion
  }):Timeout(22000)
```
might be more accurate

Reply:
This comment applies to all other scripts as well